apps/api/v2/src/modules/organizations/repositories/organizations-membership.repository.ts

```diff
@@ -0,0 +1,50 @@
+import { CreateOrgMembershipDto } from "@/modules/organizations/inputs/create-organization-membership.input";
+import { PrismaReadService } from "@/modules/prisma/prisma-read.service";
+import { PrismaWriteService } from "@/modules/prisma/prisma-write.service";
+import { Injectable } from "@nestjs/common";
+
+@Injectable()
+export class OrganizationsMembershipRepository {
+  constructor(private readonly dbRead: PrismaReadService, private readonly dbWrite: PrismaWriteService) {}
+
+  async findOrgMembership(organizationId: number, membershipId: number) {
+    return this.dbRead.prisma.membership.findUnique({
+      where: {
+        id: membershipId,
+        teamId: organizationId,
+      },
+    });
+  }
+
+  async deleteOrgMembership(organizationId: number, membershipId: number) {
+    return this.dbWrite.prisma.membership.delete({
+      where: {
+        id: membershipId,
+        teamId: organizationId,
+      },
+    });
+  }
+
+  async createOrgMembership(organizationId: number, data: CreateOrgMembershipDto) {
+    return this.dbWrite.prisma.membership.create({
+      data: { ...data, teamId: organizationId },
+    });
+  }
+
+  async updateOrgMembership(organizationId: number, membershipId: number, data: CreateOrgMembershipDto) {
+    return this.dbWrite.prisma.membership.update({
+      data: { ...data },
+      where: { id: membershipId, teamId: organizationId },
+    });
+  }
+
+  async findOrgMembershipsPaginated(organizationTeamsIds: number[], skip: number, take: number) {
+    return this.dbRead.prisma.membership.findMany({
+      where: {
+        teamId: { in: organizationTeamsIds },
+      },
+      skip,
+      take,
+    });
+  }
+}
```

Comment:
findOrgTeams is not necessary, findOrgMemberships is currently get the memberships of the teams of an org, not the memberships of the org, which is confusing, we should make another PR with another controller to handle memberships of the teams within an org

```ts

 async findOrgMembershipsPaginated(organizationId: number, skip: number, take: number) {
    return this.dbRead.prisma.membership.findMany({
      where: {
       teamId: organizationId,
      },
      skip,
      take,
    });
  }
  
  ```